src/remote/build_metadata_store.go

```diff
@@ -0,0 +1,78 @@
+package remote
+
+import (
+	"encoding/gob"
+	"fmt"
+	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/fs"
+	"os"
+	"path/filepath"
+)
+
+const pleaseCacheDirName = "please"
+const storeDirectoryName = "build-metadata-store"
+
+type buildMetadataStore interface {
+	StoreMetadata(key string, md *core.BuildMetadata) error
+	RetrieveMetadata(key string) (*core.BuildMetadata, error)
+}
+
+func newDirMDStore() *directoryMetadataStore {
+	userCacheDir, err := os.UserCacheDir()
+	if err != nil {
+		panic(fmt.Errorf("failed to find user cache dir for metadata store: %v", err))
+	}
+	dir := filepath.Join(userCacheDir, pleaseCacheDirName, storeDirectoryName)
```

Comment:
Might be worth putting an extra layer in here (I quite like a two-letter prefix) so you don't just have one directory with millions of entries in it... (you'd at  least have 1/256th of them each)

Reply:
I *could* use the targets package and name?